pkg/common/config/config.go

```diff
@@ -262,10 +262,11 @@ func GetCnsconfig(cfgPath string) (*Config, error) {
 	var cfg *Config
 	//Read in the vsphere.conf if it exists
 	if _, err := os.Stat(cfgPath); os.IsNotExist(err) {
+		klog.V(2).Infof("Could not stat %s, reading config params from env", cfgPath)
 		// config from Env var only
 		cfg = &Config{}
-		if err := FromEnv(cfg); err != nil {
-			klog.Errorf("Error reading vsphere.conf\n")
+		if fromEnvErr := FromEnv(cfg); fromEnvErr != nil {
+			klog.Errorf("Failed to get config params from env. Err: %v", fromEnvErr)
 			return cfg, err
```

Comment:
Shouldn't this now be `fromEnvErr` instead of `err` that is returned?

Reply:
Not quite. I want to catch `os.IsNotExist(err)` which is thrown by `Stat()`. 
`FromEnv()` returns errors like `ErrMissingVCenter` that the calling function to `GetCnsconfig` wouldn't be interested in.